src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/DocumentPullDiagnosticsEndpoint.cs

```diff
@@ -42,6 +42,12 @@ internal class DocumentPullDiagnosticsEndpoint : IRazorRequestHandler<VSInternal
     public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, VSInternalClientCapabilities clientCapabilities)
     {
         serverCapabilities.SupportsDiagnosticRequests = true;
+        if (serverCapabilities.DiagnosticProvider is null)
+        {
+            serverCapabilities.DiagnosticProvider = new VSInternalDiagnosticOptions();
+        }
```

Comment:
```suggestion
		serverCapabilities.DiagnosticProvider ??= new VSInternalDiagnosticOptions()
```

Reply:
??= works nicely here; the method is now half the original length.